force_wfmanager/ui/setup/setup_pane.py

```diff
@@ -33,201 +32,235 @@ class SetupPane(TraitsTaskPane):
-                            width=825
+                            visible_when="selected_model is not None"
                         ),
-                        springy=True,
+                        # Remove Buttons
+                        HGroup(
+                            UItem('remove_entity_btn', label='Delete'),
+                        ),
+                        label="Item Details",
+                        visible_when="instance_view_visible",
+                        show_border=True,
                     ),
-                    HGroup(
-                        UItem(
-                            'add_new_entity_btn',
-                            editor=ButtonEditor(
-                                label_value='add_new_entity_label'
+                    # Factory View
+                    VGroup(
+                        HGroup(
+                            UItem(
+                                "object.system_state.entity_creator",
+                                editor=InstanceEditor(),
+                                style="custom",
+                                visible_when="object.system_state"
+                                             ".entity_creator "
+                                             "is not None",
+                                width=825
                             ),
-                            enabled_when='enable_add_button',
-                            visible_when="selected_factory_name != 'None'",
-                            springy=True
+                            springy=True,
                         ),
-                        # Remove Buttons
-                        UItem(
-                            'remove_entity_btn',
-                            label='Delete Layer',
-                            visible_when=(
-                                "selected_factory_name == 'Data Source'"
+                        HGroup(
+                            UItem(
+                                'add_new_entity_btn',
+                                editor=ButtonEditor(
+                                    label_value='add_new_entity_label'
+                                ),
+                                enabled_when='enable_add_button',
+                                visible_when="object.system_state"
+                                             ".selected_factory_name "
+                                             "!= 'None'",
+                                springy=True
                             ),
+                            # Remove Buttons
+                            UItem(
+                                'remove_entity_btn',
+                                label='Delete Layer',
+                                visible_when=(
+                                    "object.system_state"
+                                    ".selected_factory_name "
+                                    "== 'Data Source'"
```

Comment:
Can this (and all the remaining long conditions) be made into a property, to complete the refactoring?

Reply:
Done